src/sonic-config-engine/minigraph.py

```diff
@@ -413,6 +428,81 @@ def parse_deviceinfo(meta, hwsku):
                 port_speeds[port_alias_map.get(alias, alias)] = speed
     return port_speeds, port_descriptions
 
+# Special parsing for spine chassis frontend 
+def parse_spine_chassis_fe(results, vni, lo_intfs, phyport_intfs, pc_intfs, pc_members, devices):
+    chassis_vnet ='VnetFE'
+    chassis_vxlan_tunnel = 'TunnelInt'
+    chassis_vni = vni
+
+    # Vxlan tunnel information
+    lo_addr = '0.0.0.0'
+    for lo in lo_intfs:
+        lo_network = ipaddress.IPNetwork(lo[1])
+        if lo_network.version == 4:
+            lo_addr = str(lo_network.ip)
+            break        
+
+    results['VXLAN_TUNNEL'] = {chassis_vxlan_tunnel: {
+        'source_ip': lo_addr
+    }}
+
+    # Vnet information
+    results['VNET'] = {chassis_vnet: {
+        'vxlan_tunnel': chassis_vxlan_tunnel,
+        'vni': chassis_vni
+    }}
+
+    # Find physical L3 interfaces that should be enslaved to Vnet
+    for intf in phyport_intfs:
+        if isinstance(intf, tuple) == False:
+            continue 
+
+        # intf = (intf name, IP prefix)
+        intf_name = intf[0]
+        neighbor_router = results['DEVICE_NEIGHBOR'][intf_name]['name']
+            
+        # If the neighbor router is an external router 
+        if devices[neighbor_router]['type'] != chassis_backend_role:
+
+            # Enslave the interface to a Vnet
+            if intf_name in phyport_intfs:
+                phyport_intfs[intf_name] = {'vnet_name': chassis_vnet}
+            else:
+                print >> sys.stderr, 'Warning: cannot find the key %s' % (intf_name) 
+
+    # Find port chennel interfaces that should be enslaved to Vnet
+    for pc_intf in pc_intfs:
+        if isinstance(pc_intf, tuple) == False:
+            continue 
+
+        # Get port channel interface name
+        # pc intf = (pc intf name, IP prefix)
+        pc_intf_name = pc_intf[0]
+
+        # Get an interface that is enslaved to this port channel 
+        intf_name = None 
+        for pc_member in pc_members:
+            if isinstance(pc_member, tuple) and pc_member[0] == pc_intf_name:
```

Comment:
why do we need to check if pc_member is a tuple or not?

Reply:
https://github.com/Azure/sonic-buildimage/pull/3100/files#diff-b589ff3cc24987df1f37e0c79272bf08R262

Reply:
@lguohan  Got it. I will update it soon.

Reply:
@lguohan I check https://github.com/Azure/SONiC/wiki/Configuration#port-channel-member. It seems that pc_member must be a tuple (port channel, physical interface). So I remove check here.